### PR TITLE
[SMTChecker] Support unary inc/dec for array/mapping access

### LINKED ISSUE #6401: [SMTChecker] Support unary increment/decrement for IndexAccess

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes #6401

### CODE REVIEW

File: libsolidity/formal/SMTChecker.cpp [hunk 1]
```diff
@@ -485,21 +485,32 @@ void SMTChecker::endVisit(UnaryOperation const& _op)
 
 		solAssert(isInteger(_op.annotation().type->category()), "");
 		solAssert(_op.subExpression().annotation().lValueRequested, "");
-		if (Identifier const* identifier = dynamic_cast<Identifier const*>(&_op.subExpression()))
+		if (
+			dynamic_cast<Identifier const*>(&_op.subExpression()) ||
```
Comment (ekpyron) [inline libsolidity/formal/SMTChecker.cpp]: I wonder whether avoiding to duplicate two lines below is worth adding 7 lines for the case distinction inside this combined if... don't you think splitting the two cases might be easier to read :-)?
In any case, it should be fine as it is as well.

**Review by ekpyron — DISMISSED** [review]
Not sure whether it wouldn't be nicer to split the cases, but looks fine either way.

File: libsolidity/formal/SMTChecker.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/formal/SMTChecker.cpp]: Can also do that. I basically copied this from the `assignment` visitor which used to do it separate and was recently combined :p

Comment (ekpyron) [reply libsolidity/formal/SMTChecker.cpp]: Fine with me either way :-).

Comment (leonardoalt) [reply libsolidity/formal/SMTChecker.cpp]: Updated

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Merging without test run...

**Pull request merged**